From 0ffc3b2c1faf5bcac7ff2c0301ec40b8622f9a04 Mon Sep 17 00:00:00 2001
From: Quentin Casasnovas <quentinx.casasnovas@intel.com>
Date: Mon, 7 Jan 2013 15:32:59 +0100
Subject: [PORT FROM MAIN] wl12xx: Dead code removal in suspend path

BZ: 75240

This patch removes the unnecessary code for a cleaner driver and easier
reviews of the code in the suspend path.

Change-Id: Ie230c76cb6774f2f8b97bc0e1ddfc312be136ba8
Orig-Change-Id: Iade235cdf1585dbc33f01cd7dd7acb3840d48632
Signed-off-by: Quentin Casasnovas <quentinx.casasnovas@intel.com>
Reviewed-on: http://android.intel.com:8080/85336
Reviewed-by: De Cesco, Jonathan <jonathan.de.cesco@intel.com>
Tested-by: Zurmely, PierreX <pierrex.zurmely@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 wl12xx-compat/drivers/net/wireless/wl12xx/main.c |   12 ++----------
 1 files changed, 2 insertions(+), 10 deletions(-)

diff --git a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
index 511bf8b..d6bb73d 100644
--- a/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
+++ b/wl12xx-compat/drivers/net/wireless/wl12xx/main.c
@@ -2316,26 +2316,18 @@ static int wl1271_op_resume(struct ieee80211_hw *hw)
 {
 	struct wl1271 *wl = hw->priv;
 	struct wl12xx_vif *wlvif;
-	unsigned long flags;
 	bool run_irq_work = false, pending_recovery;
 	int ret = 0;
 
 	wl1271_debug(DEBUG_MAC80211, "mac80211 resume wow=%d",
 		     wl->wow_enabled);
 
-	/* Request from mac80211 to go through regular reset on wakeup */
-	if (!wl->wow_enabled)
-		ret = 1;
+	/* This should not be possible as it is set by wl1271_op_suspend() */
+	WARN_ON(!wl->wow_enabled);
 
-	/*
-	 * re-enable irq_work enqueuing, and call irq_work directly if
-	 * there is a pending work.
-	 */
-	spin_lock_irqsave(&wl->wl_lock, flags);
 	clear_bit(WL1271_FLAG_SUSPENDED, &wl->flags);
 	if (test_and_clear_bit(WL1271_FLAG_PENDING_WORK, &wl->flags))
 		run_irq_work = true;
-	spin_unlock_irqrestore(&wl->wl_lock, flags);
 
 	mutex_lock(&wl->mutex);
 
-- 
1.7.4.1

